test/jit/test_types.py

```diff
@@ -244,6 +244,32 @@ def forward(self) -> int:
         with self.assertRaisesRegexWithHighlight(RuntimeError, r"attribute was ignored during compilation", "self.sub"):
             scripted_mod = torch.jit.script(mod)
 
+
+    def test_ignoring_fn_with_nonscriptable_types(self):
+        class CFX(object):
+            def __init__(self, a: List[torch.Tensor]) -> None:
+                self.a = a
+
+            def forward(self, x: torch.Tensor) -> torch.Tensor:
+                return torch.sin(x)
+
+            @torch.jit._drop
+            def __iter__(self) -> Iterator[torch.Tensor]:
+                return iter(self.a)
+
+            @torch.jit._drop
+            def __fx_create_arg__(self, tracer: torch.fx.Tracer) -> torch.fx.node.Argument:
+                # torch.fx classes are not scriptable
+                return tracer.create_node(
+                    "call_function",
+                    CFX,
+                    args=(tracer.create_arg(self.features),),
+                    kwargs={},
+                )
+
+        torch.jit.script(CFX)
```

Comment:
can you add an export + import test here? [example](https://github.com/pytorch/pytorch/blob/b179a097ea03154de6d21abbbc7436adec0d3f73/test/jit/test_save_load.py#L231-L234).

I did notice that you can still call `__fx_create_arg__` on the scripted class. Might be good to document this (or disable it if you know how, but I'm not exactly sure right now what the best way to do this is)